pennylane_lightning/src/Gates.cpp

```diff
@@ -61,6 +85,10 @@ const vector<CplxType> Pennylane::XGate::matrix{
     0, 1,
     1, 0 };
 
+void Pennylane::XGate::applyKernel(CplxType* state, vector<size_t>& indices, vector<CplxType>& v) {
```

Comment:
I'm okay with adding the `const`; I'm generally not a fan of removing variable names from method signatures, but I guess opinions may differ on this one--if nothing else, its making it clear that its overriding the same `virtual` method.

Reply:
Got it! A reason to make `v` unnamed would be that while reading, having the name `v` was somewhat confusing, as one would expect that to be used within the function (as is in some of them).

Reply:
True; although I think that's indication we should rework the API (i.e. we shouldn't be passing in parameters we don't need if we don't have to). As mentioned in the PR description, its probably beneficial, although it will unfortunately result in some measure of code duplication.

Reply:
Or if branching off would result in a performance degradation, how about incorporating the internal loop in `applyKernel` or creating another member function that implements the internal loop and depends on `applyKernel`? That way `inputVector` would only be created if used.

Reply:
That would be ideal. However, ultimately I think we want to inline `applyKernel`. And for that to happen I think we need to duplicate the outer loop for every subclass, because of the requirement to be able to statically infer which method is being called. Again, probably beneficial in the long run, but just a bit painful to look at.